daemon-scheduler/pkg/deployment/environment_test.go

```diff
@@ -185,6 +186,139 @@ func (suite *EnvironmentTestSuite) TestGetEnvironment() {
+	assert.True(suite.T(), ok, "Expected NotFoundError")
+}
+
+func (suite *EnvironmentTestSuite) TestDeleteEnvironment() {
+	environment, err := types.NewEnvironment("TestDeleteEnvironment", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(environment, nil)
+	suite.environmentStore.EXPECT().DeleteEnvironment(suite.ctx, *environment).
+		Return(nil).Times(1)
+
+	err = suite.environment.DeleteEnvironment(suite.ctx, environment.Name)
+	assert.Nil(suite.T(), err, "Unexpected error")
+}
+
+func (suite *EnvironmentTestSuite) TestDeleteEnvironmentReturnsError() {
+	environment, err := types.NewEnvironment("TestDeleteEnvironmentReturnsError", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(environment, nil)
+	err = errors.New("Error calling DeleteEnvironment")
+	suite.environmentStore.EXPECT().DeleteEnvironment(suite.ctx, *environment).
+		Return(err).Times(1)
+
+	observedErr := suite.environment.DeleteEnvironment(suite.ctx, environment.Name)
+	assert.Exactly(suite.T(), err, errors.Cause(observedErr))
+}
+
+func (suite *EnvironmentTestSuite) TestDeleteEnvironmentEmptyName() {
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, gomock.Any()).
+		Times(0)
+	suite.environmentStore.EXPECT().DeleteEnvironment(suite.ctx, gomock.Any()).
+		Times(0)
+
+	_, ok := suite.environment.DeleteEnvironment(suite.ctx, "").(types.BadRequestError)
+	assert.True(suite.T(), ok, "Expecting BadRequestError")
+}
+
+func (suite *EnvironmentTestSuite) TestDeleteEnvironmentGetEnvironmentReturnsError() {
+	environment, err := types.NewEnvironment("TestDeleteEnvironmentReturnsError", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+
+	err = errors.New("Error calling GetEnvironment")
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(nil, err)
+	suite.environmentStore.EXPECT().DeleteEnvironment(suite.ctx, *environment).
+		Times(0)
+
+	observedErr := suite.environment.DeleteEnvironment(suite.ctx, environment.Name)
+	assert.Exactly(suite.T(), err, errors.Cause(observedErr))
+}
+
+func (suite *EnvironmentTestSuite) TestDeleteEnvironmentGetEnvironmentReturnsEmpty() {
+	environment, err := types.NewEnvironment("TestDeleteEnvironmentGetEnvironmentReturnsEmpty", "td-arn", "cluster-arn")
+	assert.Nil(suite.T(), err, "Unexpected error when creating environment")
+
+	suite.environmentStore.EXPECT().GetEnvironment(suite.ctx, environment.Name).
+		Return(nil, nil)
+	suite.environmentStore.EXPECT().DeleteEnvironment(suite.ctx, *environment).
+		Times(0)
+
+	observedErr := suite.environment.DeleteEnvironment(suite.ctx, environment.Name)
+	assert.Nil(suite.T(), observedErr, "Unexpected error")
```

Comment:
I'm not sure I understand. If the environment requested to be deleted, it's not an error?